kolena/dataset/dataset.py

```diff
@@ -99,7 +99,11 @@ def _normalize_url(x: str) -> str:
 
 
 def _infer_datatype_value(x: str) -> str:
-    url = _normalize_url(x)
+    try:
+        url = _normalize_url(x or "")
+    except Exception:
+        return DatapointType.TABULAR.value
```

Comment:
i think we can check the type of x, if not string just fallback to `DatapointType.TABULAR` to make the logic cleaner

Reply:
Given that this is leveraging library, and we don't have all possible scenarios that could fail, I'm using try/exception over trying to if-condition all fail scenarios.

Reply:
if we use try-exception at here, assuming the type of x could be `Any`
- we will need to update `_normalize_url` to accept Any
- `urlparse` only accept string, then we will doing the type checking in `_normalize_url`

that's why i suggest to do the type checking at `_infer_datatype_value` level

Reply:
updated as suggested